Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp

```diff
@@ -333,14 +333,30 @@ bool CStaticFunctionDefinitions::DestroyElement(CElement* pElement)
     {
         return false;
     }
+    // Its team trigger onPlayerTeamChange for each player in the team
+    else if (iType == CElement::TEAM) 
+    {
+        CTeam* pTeam = static_cast<CTeam*>(pElement);
+
+        auto iterBegin = pTeam->PlayersBegin();
+        auto iterEnd = pTeam->PlayersEnd();
+
+        for (auto iter = iterBegin; iter != iterEnd; ++iter){
+            CPlayer* player = *iter;
+            CLuaArguments arguments;
+            arguments.PushNil(); // Current team going to delete so return nil
```

Comment:
damn, I don't know about this one. I'd love to have the real team element there. Even though it's about to be deleted.

CallEvent are synchronous so I think we should still send it, all the event handlers will do what they have to do with the team element being destroyed and I think it will do it.

PS: If anyone sees a reason not to send the team element being destroyed at that stage, please add your input.

Reply:
at first it was team element, but i thought it will be better to return nil, but i think return team element its actually better